hazelcast/src/main/java/com/hazelcast/instance/impl/HazelcastBootstrap.java

```diff
@@ -168,18 +175,38 @@ public static synchronized void executeJar(@Nonnull Supplier<HazelcastInstance>
             String[] jobArgs = args.toArray(new String[0]);
             // upcast args to Object so it's passed as a single array-typed argument
             main.invoke(null, (Object) jobArgs);
-            awaitJobsStarted();
+
+            // Method is synchronized so, it is safe to access submittedJobs array in BootstrappedJetProxy
+            awaitJobsStarted(calledByMember);
+
         } finally {
-            HazelcastInstance remembered = HazelcastBootstrap.supplier.remembered();
-            if (remembered != null) {
-                try {
-                    remembered.shutdown();
-                } catch (Throwable t) {
-                    System.err.println("Shutdown failed with:");
-                    t.printStackTrace();
+            // HazelcastInstance should be closed if called by client side
+            // HazelcastInstance should not be closed if called by member side
+            if (!calledByMember) {
+                HazelcastInstance remembered = HazelcastBootstrap.supplier.remembered();
+                if (remembered != null) {
+                    try {
+                        remembered.shutdown();
+                    } catch (Throwable t) {
+                        System.err.println("Shutdown failed with:");
```

Comment:
please use logger for that

Reply:
This code is used by the client side. There is no logger on the hz-client. That's why they have used system.err in all over this class.

Reply:
I mean I did not put system.out calls into this class. They were there already for the reason specified above

Reply:
Right, I forgot, sorry